config/common.php

```diff
@@ -22,12 +23,16 @@
     ],
     RuleHandlerResolverInterface::class => SimpleRuleHandlerContainer::class,
     'yii.validator.categorySource' => [
-        'definition' => static function (IdMessageReader $idMessageReader) use ($params): CategorySource {
-            return new CategorySource(
-                $params['yiisoft/validator']['translation.category'],
-                $idMessageReader,
-                extension_loaded('intl') ? new IntlMessageFormatter() : new SimpleMessageFormatter(),
-            );
+        'definition' => static function () use ($params): CategorySource {
+            $reader = class_exists(MessageSource::class)
+                ? new MessageSource(dirname(__DIR__) . '/messages')
+                : new IdMessageReader(); // @codeCoverageIgnore
+
+            $formatter = extension_loaded('intl')
+                ? new IntlMessageFormatter()
+                : new SimpleMessageFormatter(); // @codeCoverageIgnore
```

Comment:
Why ignore it from coverage? The test is still present:

https://github.com/yiisoft/validator/blob/d9647f35475eaa0e6081af80fe49decfce86f1f8/tests/ConfigTest.php

Reply:
I'm make this in separate PR.

Reply:
Current CI configuration incorrect. In PHP 8.0 all tests run with intl extension. I'm fixed this in another PR.